src/rng.rs

```diff
@@ -48,15 +43,18 @@ impl<R: CryptoRng> rand_crypto::CryptoRng for RngCompat<R> {}
 // Rng implementation used in production. Uses `OsRng` for maximum cryptographic security.
 #[cfg(not(any(test, feature = "mock_base")))]
 mod implementation {
-    pub use rand::OsRng as MainRng;
+    // Parsec has a very old rand dependecy. This osRng is required to interact wiht that, until
+    // parsec upgrades
```

Comment:
Why is that?

Reply:
Also these `RngCompat` types are pretty trivial, so you can you one for now that implement I guess Rng for the right rand crate version. and then call `Box::new(ParsecRngCompat(rng::new_from(rng)))`

Reply:
I think this is something I do need to do, I fought this error for many hours this weekend. It seems newer versions of OsRng impl traits the old one does not have. So ends up in a cycle of messiness.